### ISSUE TITLE
Not working with `@cycjimmy/semantic-release-action`

### ISSUE THREAD

**Issue by felixjb** [felipecrs/semantic-release-vsce#295]
When the semantic release action is executed using the vsce plugin in my repo, [Testify](https://github.com/felixjb/testify), i get the following error:

```
[8:23:46 PM] [semantic-release] › ✖  EINVALIDVSCETOKEN Invalid vsce token. Additional information:

Error: Command failed with ENOENT: vsce verify-pat
spawn vsce ENOENT
Error: AggregateError: 
    SemanticReleaseError: Invalid vsce token. Additional information:
    Error: Command failed with ENOENT: vsce verify-pat
    spawn vsce ENOENT
        at module.exports (/home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/semantic-release-vsce/lib/verify-auth.js:14:11)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
        at async module.exports (/home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/semantic-release-vsce/lib/verify.js:8:3)
        at async verifyConditions (/home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/semantic-release-vsce/index.js:10:3)
        at async validator (/home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/semantic-release/lib/plugins/normalize.js:34:24)
        at async /home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/semantic-release/lib/plugins/pipeline.js:37:34
        at async Promise.all (index 0)
        at async next (/home/runner/work/_actions/cycjimmy/semantic-release-action/v3/node_modules/p-reduce/index.js:16:18)
```

I am using the GitHub secrets to store my Personal Access Token and I double checked that the token is really there. 

I also double checked the permissions of said token in the Azure Portal and it has the necessary access specified in the VSCode [documentation](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#get-a-personal-access-token) 

When I run the same command as the plugin locally, I get the following result:

```
The Personal Access Token verification succeeded for the publisher 'felixjb'.
```

Is this a bug? Am I doing anything wrong?

**Comment by felipecrs** [comment]
This is very weird. Perhaps you may have more luck turning debug logs on?

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging

**Comment by felipecrs** [comment]
One suggestion is to run:

```
vsce verify-pat
```

Directly in your workflow.

**Comment by felixjb** [comment]
Alright, I'll do that right now!

**Comment by felipecrs** [comment]
Actually, nevermind. I think the issue is due to https://github.com/felipecrs/semantic-release-vsce/pull/294 maybe. I'll check it further.

**Comment by felipecrs** [comment]
As a temporary workaround, try to run with the last but one version of the plugin. If it works, please let me know to confirm the issue.

**Comment by felixjb** [comment]
So, I ran the workflow again with the debug logs on and it seems like the variable is there:

```
##[debug]Evaluating: secrets.VSCE_TOKEN
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'VSCE_TOKEN'
##[debug]=> '***'
##[debug]Result: '***'
```

**Comment by felixjb** [comment]
Sorry, I didn't get that. I should try again with the second to last version of the plugin?

**Comment by felipecrs** [comment]
Yes, exactly 5.0.14.

**Comment by felixjb** [comment]
Alright. I'll do that now

**Comment by felixjb** [comment]
@felipecrs Do I need to update the version in the package.json too if I'm running the workflow on GitHub Actions?

**Comment by felixjb** [comment]
> As a temporary workaround, try to run with the last but one version of the plugin. If it works, please let me know to confirm the issue.

@felipecrs Nope, didn't work either

**Comment by felipecrs** [comment]
> @felipecrs Do I need to update the version in the package.json too if I'm running the workflow on GitHub Actions?

I don't know... never used the semantic-release-action myself, I just call semantic-release directly. Like in [vscode-shellcheck](https://github.com/vscode-shellcheck/vscode-shellcheck/blob/master/.github/workflows/ci.yaml#L49).

**Comment by felixjb** [comment]
I just tried using a full access token and even that didn't work. 

Do you think it's possible that because I didn't register a `NPM_TOKEN` for semantic release, as I am using the `private: true` option for not publishing to npm registry, it is messing up with the `VSCE_PAT`?

**Comment by felipecrs** [comment]
No, I don't think so. Can you try without using the semantic-release action?

I'm trying to replicate the issue here, but without much success.

**Comment by felixjb** [comment]
Sure. I'll try it now

**Comment by felixjb** [comment]
It seems like it works without the action:

![image](https://user-images.githubusercontent.com/16679401/171759746-2a1b6b34-6af5-4431-94b2-683d85954468.png)

**Comment by felipecrs** [comment]
I meant this: https://github.com/felixjb/testify/pull/97

**Comment by felixjb** [comment]
Oh, I see! Wouldn't it be better to make the plugin work with the action?

**Comment by felipecrs** [comment]
Yes, but first we need to make sure that's the issue indeed. Right?

**Comment by felixjb** [comment]
Fair enough. Can we continue on this tomorrow then? Haha

**Comment by felipecrs** [comment]
What's happening is the following:

- The plugin fails to call vsce (we use execa for that) with ENOENT (file not found)
- This probably means that vsce isn't in the PATH environment variable.

Probably something different happens in the action which makes the PATH variable not be filled properly (i.e. with node_modules/.bin on it).

**Comment by felipecrs** [comment]
> Fair enough. Can we continue on this tomorrow then? Haha

@felixjb lol, sure. I'm on vacation anyways!

**Comment by felixjb** [comment]
@felipecrs It seems like running the semantic release directly worked! Now I have some other issues that are unrelated. 

So, that is it: the plugin cannot verify the PAT when using the GitHub actions at this time. Maybe try explicitly passing the PAT and the publisher to the command can help?

**Comment by felipecrs** [comment]
> So, that is it: the plugin cannot verify the PAT when using the GitHub actions at this time. Maybe try explicitly passing the PAT and the publisher to the command can help?

I guess the correct assessment is: the plugin cannot call the `vsce` binary when running upon @cycjimmy/semantic-release-action.

---

@cycjimmy do you have any clue on why this is happening? The plugin does not do anything special, it just calls the `vsce` binary [here](https://github.com/felipecrs/semantic-release-vsce/blob/master/lib/verify-auth.js#L12), the same way other semantic-release plugins do.

The catch is that `vsce` comes from the NPM dependency, therefore, it's executable is on `node_modules/.bin`.

**Comment by felipecrs** [comment]
I believe using execa's `preferLocal` option, together with #338 may fix this issue. I'll check later.

**Comment by felipecrs** [comment]
Hey @felixjb, I published [v5.2.2](https://github.com/felipecrs/semantic-release-vsce/releases/tag/v5.2.2) which contains a potential fix for this issue. Would you mind testing it? Thanks in advance.

**Comment by felipecrs** [comment]
I'm closing this issue as I believe it's fixed. If someone notices the opposite, please comment here and I can reopen it.
